### ISSUE TITLE
`regex-expr` and `contain-expr` are missing lhs

### ISSUE THREAD

**Issue by cabo** [ietf-wg-jsonpath/draft-ietf-jsonpath-base#120]
While this may be a bit premature as long as we don't know what the rhs of contain-expr should be, it is missing its lhs.

```
contain-expr = in-op container
in-op        = " in "                                 ; in operator
```

*What* is in the container?

**Comment by cabo** [comment]
Oh, and the same with `regex-expr`:

```
regex-expr   = regex-op regex
```

**Comment by goessner** [comment]
Well ... good point. Starting with these examples:
```
@.color in ["red","green","blue"]
@.color in $.colors
$.color in @.colors
"color" in @.colors
"color" in @
@.index in [2,4,7]
@.index in $.indices
$.index in @.indices
42 in @.indices
42 in @
```
* lhs is restricted to a primitive value (possibly only number and string). lhs holding a complex value will fail, i.e. no selection occurs.
* rhs must be resolved to a complex value (array or object).  rhs holding a simple value will fail, i.e. no selection occurs.

Given this we have something like this ...
```
contain-expr = (rel-path-val / json-path / number / string-literal) in-op container
container =  (rel-path-val / json-path / array-literal)
```
... as a starting proposal.

**Comment by cabo** [comment]
(From 2012-09-17 interim:)
So we would have syntax for array literals composed of primitive values only (subset of JSON array literals); other JSON array literals cause a syntax error.  (Empty array literals should be allowed to simplify code generation.)
A LHS evaluating to a complex value would not match.
The path on the RHS may also yield an array, we would match against every single element of that like we would match against a literal array.
Example: `"color" in @.colors`

tending towards consensus on:
`@.color in $.colors`
if $.colors is an array of structured values, @.color is a single structured value?
➔ only match primitive values from the LHS (counterintuitive!?)
`@.color` = [128, 255, 17] ➔ no selection 
even if `$.color` = [[128, 255, 17], [1, 2, 3]]
(we cannot avoid structured data to turn up in a query, but we don't match)
Avoid deep equals!  (We already avoid it in simple comparisons.)

Discuss:
`@.color in $..allowed_color`
We make sure that the paths on the LHS only yield a single node; the paths on the RHS may yield multiple nodes and the nodelist is examined like a JSON array would be.
\[How do we distinguish these cases?]
➔ disallow paths that can yield multiple nodes on the in-op RHS as well.

**Comment by goessner** [comment]
Reminder: lhs of `regex-op` ...

From
```
regex-expr   =   regex-op regex
```
to
```
regex-expr   = rel-path / json-path /                 ; path to string value
                      string-literal
                      regex-op regex
```
as a proposal.

**Comment by cabo** [comment]
Fixed in 9da13ab4

(Still needs resolution of #70 to become active.)
